agent/common.py

```diff
@@ -0,0 +1,276 @@
+"""Common utilities for the exploits."""
+
+import ipaddress
+import json
+from typing import cast, Any
+from urllib import parse
+
+import tld
+from ostorlab.agent.mixins import agent_report_vulnerability_mixin
+from ostorlab.assets import domain_name as domain_asset
+from ostorlab.assets import ipv4 as ipv4_asset
+from ostorlab.assets import ipv6 as ipv6_asset
+from ostorlab.assets import file as file_asset
+from ostorlab.assets import android_apk as android_apk_asset
+from ostorlab.assets import android_aab as android_aab_asset
+from ostorlab.assets import ios_ipa as ios_ipa_asset
+
+
+def prepare_host(host: str) -> str:
+    """Prepare the host for the request.
+    Args:
+        host: The host to prepare.
+    Returns:
+        The prepared host.
+    """
+    if _is_ipv6(host) is True:
+        return f"[{host}]"
+    return host
+
+
+def _is_ipv4(potential_ip: str) -> bool:
+    """check if the potential_ip is a valid ipv4.
+
+    Args:
+        potential_ip: string.
+
+    Returns:
+        - boolean.
+    """
+    ip, _ = _split_ipv4(potential_ip)
+    try:
+        ipaddress.IPv4Address(ip)
+        return True
+    except ValueError:
+        return False
+
+
+def _split_ipv4(potential_ip: str) -> tuple[str, str | None]:
+    """split the potential_ip to get the ip and the port if existed.
+
+    Args:
+        potential_ip: string.
+
+    Returns:
+        - ip, port.
+    """
+    ip = potential_ip
+    port = None
+    if ":" in potential_ip:
+        ip, port = potential_ip.split(":", maxsplit=1)
+    return ip, port
+
+
+def _is_ipv6(potential_ip: str) -> bool:
+    """check if the potential_ip is a valid ipv6.
+
+    Args:
+        potential_ip: string.
+
+    Returns:
+        - boolean.
+    """
+    try:
+        ipaddress.IPv6Address(potential_ip)
+        return True
+    except ValueError:
+        return False
+
+
+def build_vuln_location(
+    target_url: str | None,
+    file_path: str | None = None,
+    file_content: bytes | None = None,
+    selector: str | None = None,
+    content_url: str | None = None,
+) -> agent_report_vulnerability_mixin.VulnerabilityLocation | None:
+    """Build VulnerabilityLocation based on the asset.
+
+    Args:
+        target_url: The URL
+        file_path: The path of the file
+        file_content: The content of the file
+        selector: The message selector
+        content_url: The content url of the file
+    Returns:
+        The vulnerability location object.
+    """
+
+    asset: (
+        ipv4_asset.IPv4
+        | ipv6_asset.IPv6
+        | domain_asset.DomainName
+        | file_asset.File
+        | android_aab_asset.AndroidAab
+        | android_apk_asset.AndroidApk
+        | ios_ipa_asset.IOSIpa
+    )
+
+    if selector == "v3.asset.file.android.apk":
+        asset = android_apk_asset.AndroidApk(
+            path=file_path, content=file_content, content_url=content_url
+        )
```

Comment:
The API on RE in it's current implementation only supports the ng_assets. The mobile assets will go through the store classes with package name or bundle id.

Reply:
Addressed